grimoire/districts/district_analyze.py

```diff
@@ -0,0 +1,208 @@
+from gdpc import Block
+from ..core.maps import Map
+from gdpc.vector_tools import ivec2, ivec3
+from ..districts.district import District, SuperDistrict
+import math
+
+URBAN_SIZE = 800 #max number of urban districts
```

Comment:
This is definitely up to the behvaior we want the generator to have. Personally I would rather have the terrain decide how large the city is going to be (or some other equation we come up with based on the generation), rather than us arbitrarily limiting urban size. Therefore I just made this number bigger than it should ever be reached for now, but still left the functionality in to limit it.